### ISSUE TITLE
MILO client not working with localhost OPCUA server. Using OPC2AAS repo.

### ISSUE THREAD

**Issue by suman-v** [eclipse-basyx/basyx-applications#188]
I am trying to use Databridge functionality to connect Docker container with local opcua server running in my system. I followed all steps mentioned in this repo but still not able to connect to opcua server.
Repo: https://github.com/eclipse-basyx/basyx-applications/tree/main/opc2aas
 Please help me here. 
Thank you in advance. 
![image](https://github.com/eclipse-basyx/basyx-applications/assets/43817111/ab666eec-d1e9-4bc2-ab88-a2c14a2fa8ca)
![image](https://github.com/eclipse-basyx/basyx-applications/assets/43817111/b8364dfd-f909-4abb-9a7f-2de85d2221ac)

**Comment by aaronzi** [comment]
Thank you for raising this issue. With local milo server you mean a server running outside of docker? Have you already verified that another OPC UA client can access the milo server? Also, is the server secured in any way?

**Comment by suman-v** [comment]
Yes, I am using UAExpert to connect to OPCUA server running in my local window system and it is able to connect.

**Comment by aaronzi** [comment]
Thank you for the info. Is the DataBridge configuration autogenerated or did you manually add something to the `opcuaconsumer.json`? I am asking because the `serverPort` is not the OPC UA standard port (4840) and the `pathToService` is set to milo. Are you using milo on your local windows as OPC UA server?

**Comment by suman-v** [comment]
Yes it was autogenerated but i added "Milo" manually

**Comment by aaronzi** [comment]
Ok, thank you for the info. You should remove the `pathToService` then. Could you try without that again and send me the log for the DataBridge if it still fails?

**Comment by suman-v** [comment]
[databridge.txt](https://github.com/user-attachments/files/16081354/databridge.txt)
databridge is not starting after removing pathToService
![image](https://github.com/eclipse-basyx/basyx-applications/assets/43817111/ec9adbff-f9f4-417b-9adb-fc72730aa373)

**Comment by aaronzi** [comment]
Sorry, I think my suggestion was not phrased in an optimal way. I actually meant to do it like this:

```json
[
  {
    "uniqueId" : "source://Aliases/FindAliases/InputArguments",
    "serverUrl" : "host.docker.internal",
    "serverPort" : 4851,
    "pathToService" : "",
    "nodeInformation" : "i=23477",
    "requestPublishInterval" : 50
  }
]
```

Also what I noticed from your log is that you are using a timerconsumer. I would suggest to use `event` as trigger. This improves performance and in general works better. A route could look like this:

```json
[
  {
    "datasource" : "source://Aliases/FindAliases/InputArguments",
    "transformers" : [ "dataValueToJson", "jsonataExtractValue" ],
    "datasinks" : [ "sink://sink/id" ],
    "trigger" : "event"
  }
]
```

Also make sure to remove the `demo.txt` file from the folder where you placed your DataBridge config. The log tells me there is this unrelated text file.

**Comment by suman-v** [comment]
Okay, Thank you for your suggestions. I will try now..

**Comment by suman-v** [comment]
[databridge.txt](https://github.com/user-attachments/files/16087442/databridge.txt)
Seems like Java connector still not able to connect.

**Comment by aaronzi** [comment]
My guess would be that this is a DNS issue. The DataBridge is probably trying to reach the server by it's hostname which it can't find. What you could try is to use the IP address of the device your OPC server runs on instead of `host.docker.internal`.
